src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -227,12 +225,15 @@ void check_spatial_eq(const SpatialPooler& sp1, const SpatialPooler& sp2) {
   delete[] conCounts2;
 }
 
-void setup(SpatialPooler &sp, UInt numInputs, UInt numColumns) {
-  vector<UInt> inputDim;
-  vector<UInt> columnDim;
-  inputDim.push_back(numInputs);
-  columnDim.push_back(numColumns);
-  sp.initialize(inputDim, columnDim);
+void setup(SpatialPooler &sp, vector<UInt> inputDim, vector<UInt> columnDim, Real sparsity = 0.5f) {
+  //we are interested in the sparsity, should make it artificially high.
+  //As we added SP check that sparsity*numColumns > 0, which is correct requirement.
+  //But many tests have very small (artificial) number of columns (for convenient results),
+  //therefore the check is failing -> we must set high sparsity at initialization. 
+  EXPECT_NO_THROW(sp.initialize(inputDim, columnDim, 16u, 0.5f, true, sparsity));
```

Comment:
SP tests now use "setup()" hack to override default sparsity (reasonable, low), to one that does not trigger the SP check by using large(r) sparsity (50%)